lib/parse.js

```diff
@@ -335,7 +335,11 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         if (prefix) num = prefix + num;
         var valid = parse_js_number(num);
         if (!isNaN(valid)) {
-            return token("num", valid);
+            var tok = token("num", valid);
+            if (num.indexOf('.') >= 0) {
+                tok.literal = num;
```

Comment:
Will leave literal property as is. It works as intended.

If you want to unconditionally set the literal property for all AST_Numbers then at output time you will have to check that a `"."` exists in the literal and it's in "use asm" scope before outputting the literal.